src/TitleDetails/TitleDetailsContainer.js

```diff
@@ -34,18 +37,44 @@ const TitleDetailsContainer = ({ location, history, mutator, match }) => {
   const [poLine, setPoLine] = useState({});
   const [pieces, setPieces] = useState();
   const [locations, setLocations] = useState();
+  const [items, setItems] = useState();
+  const [requests, setRequests] = useState();
 
-  const fetchPieces = useCallback(
+  const fetchReceivingResources = useCallback(
     (lineId) => {
       setPieces();
+      setItems();
+      setRequests();
 
-      mutator.pieces.GET({
+      return mutator.pieces.GET({
         params: {
           query: `poLineId==${lineId} sortby receiptDate`,
         },
       })
-        .then(setPieces)
-        .catch(() => setPieces([]));
+        .then((fetchedPieces) => {
+          setPieces(fetchedPieces);
+
+          const itemsIds = fetchedPieces.filter(({ itemId }) => itemId).map(({ itemId }) => itemId);
+          const requestsPromise = batchFetch(mutator.requests, fetchedPieces, (piecesChunk) => {
+            const itemIdsQuery = piecesChunk
+              .filter(piece => piece.itemId)
+              .map(piece => `itemId=${piece.itemId}`)
+              .join(' or ');
+
+            return itemIdsQuery ? `(${itemIdsQuery}) and status="Open*"` : '';
+          });
+
+          return Promise.all([batchFetch(mutator.items, itemsIds), requestsPromise]);
+        })
+        .then(([pieceItems, itemRequests]) => {
+          setItems(pieceItems);
+          setRequests(itemRequests);
+        })
+        .catch(() => {
+          setPieces([]);
```

Comment:
why not to reset it before, you already call these setters without arguments

Reply:
where do you mean? it resets on the top of the method.